include/irr/asset/ICPUMeshBuffer.h

```diff
@@ -92,6 +100,16 @@ class ICPUMeshBuffer : public IMeshBuffer<ICPUBuffer, ICPUDescriptorSet, ICPURen
     {
         return m_descriptorSet.get();
     }
+	inline void setVertexAttrBuffer(core::smart_refctd_ptr<ICPUBuffer> &buffer, uint16_t attribIndex, E_FORMAT format, uint32_t stride, uint32_t offset)
+	{
+		auto &params(const_cast<SVertexInputParams&>(getPipeline()->getVertexInputParams()));
+
+		params.attributes[attribIndex].binding = attribIndex;
+		params.attributes[attribIndex].format = format;
+		params.attributes[attribIndex].relativeOffset = offset;
+		params.bindings[attribIndex].inputRate;
```

Comment:
attribute index and binding index are not same thing

Reply:
you need two functions, one for format and one for binding...